### PR TITLE
Add timeouts, max in flight, compression to API and CDN

### PR DESCRIPTION
**Pull request by MasterPtato**
Fixes RVT-3583
Fixes RVT-3584
Fixes RVT-3585
Fixes RVT-3586
Fixes RVT-3587

### CODE REVIEW

**Comment by MasterPtato** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #397** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/397?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/rivet-gg/rivet/397?utm_source=stack-comment).

**Comment by MasterPtato** [comment]
Untested, not sure how to go about testing the new timeouts

File: infra/tf/k8s_infra/traefik.tf [hunk 1]
```diff
@@ -66,9 +66,13 @@ resource "helm_release" "traefik" {
 		additionalArguments = [
 			"--providers.http.endpoint=http://rivet-api-route.rivet-service.svc.cluster.local/traefik/config/core?token=${module.traefik_secrets.values["rivet/api_route/token"]}",
 			"--providers.http.pollInterval=2.5s",
-			# See docs/infrastructure/API_TIMEOUTS.md
+			# See docs/infrastructure/TIMEOUTS.md
```
Comment (NathanFlurry) [inline infra/tf/k8s_infra/traefik.tf:69]: Can we add a lower timeout on the ATS service specifically? Not sure how this converts to the CRD: https://doc.traefik.io/traefik/routing/services/#forwardingtimeouts

File: svc/api/route/src/route/traefik/core.rs [hunk 2]
```diff
@@ -259,9 +260,9 @@ fn register_namespace(
 	let router_middlewares_base = vec![
 		// TODO: Add back
 		// "cdn-rate-limit".into(),
```
Comment (NathanFlurry) [inline svc/api/route/src/route/traefik/core.rs:262]: Remove the TODO. Rate limit will be done by CF.

File: lib/api-helper/build/src/macro_util.rs [hunk 3]
```diff
@@ -290,16 +289,16 @@ pub async fn __with_ctx<A: auth::ApiAuth + Send>(
 		// Otherwise, we use the cf-connecting-ip header since that's the recommended header to
 		// use by Cloudflare.
 		let remote_address_str =
-			if rivet_util::env::dns_provider() == Some("cloudflare") && !not_using_cloudflare {
+			if rivet_util::env::dns_provider() == Some("cloudflare") {
 				__deserialize_header::<String, _>(&request, "cf-connecting-ip")?
 			} else {
 				// TODO: This should be a comma separated list of IPs
 				// TODO: Test this is correct
```
Comment (NathanFlurry) [inline lib/api-helper/build/src/macro_util.rs:296]: If this is tested manually E2E, remove TODO

**Review by NathanFlurry — CHANGES_REQUESTED** [review]

**Review by NathanFlurry — COMMENTED** [review]
bolt check failing too

**Review by NathanFlurry — APPROVED** [review]

**Pull request merged**